### PR TITLE
add `goss.hcl`

### PR DESCRIPTION
**Pull request by MatthewDolan**

### CODE REVIEW

File: goss.hcl [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      to = "${root}/goss.exe"
+    }
+  }
+}
+
+platform "darwin" "arm64" {
+  vars = {
+    "arch_": "amd64",
+  }
+}
```
Comment (MatthewDolan) [inline goss.hcl:24]: The assets on the latest goss release (https://github.com/goss-org/goss/releases/tag/v0.3.21) does not have a `darwin` `arm64` release.

I looked around at a few other packages and found at least one example of using `amd64` when `arm64` was not present.

https://github.com/cashapp/hermit-packages/blob/cc23198eeed31c9b92de117d989b25b846541ee8/atlas.hcl#L11

My understanding is that `arm64` is M1/M2 macs and that there is a program to allow running amd64 binaries on that machine: https://en.wikipedia.org/wiki/Rosetta_(software).

**Review by alecthomas — APPROVED** [review]

**Comment by alecthomas** [comment]
Nice, LGTM! Mind running `hermit manifest add-digests goss.hcl`?

**Comment by MatthewDolan** [comment]
Ran the command. Thanks for the review, @alecthomas.

**Comment by alecthomas** [comment]
I think you'll need to rebase this against latest master if you want the tests to pass.

**Pull request merged**